core/src/test/java/kafka/test/junit/ZkClusterInvocationContext.java

```diff
@@ -106,6 +106,7 @@ public List<Extension> getAdditionalExtensions() {
             },
             (AfterTestExecutionCallback) context -> clusterShim.stop(),
             new ClusterInstanceParameterResolver(clusterShim),
+            new GenericParameterResolver<>(clusterShim, ZkClusterInstance.class),
```

Comment:
This change involve the conflicts. For another, it seems to me this could encourage developers to use specific type of `ClusterInstance`. I agree there are some test cases requiring the specific cluster type. They can cast the `ClusterInstance` to either `zk` and `Raft` so it seems to me enabling injection is overkill.

Reply:
Fair enough. I'll remove the new injection providers and just cast.